plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/split/HudiSplitFactory.java

```diff
@@ -44,57 +42,74 @@ public HudiSplitFactory(
         this.hudiSplitWeightProvider = requireNonNull(hudiSplitWeightProvider, "hudiSplitWeightProvider is null");
     }
 
-    public Stream<HudiSplit> createSplits(List<HivePartitionKey> partitionKeys, HudiFileStatus fileStatus)
+    public List<HudiSplit> createSplits(List<HivePartitionKey> partitionKeys, HudiFileStatus fileStatus)
     {
-        List<FileSplit> splits;
         try {
-            splits = createSplits(fileStatus);
-        }
-        catch (IOException e) {
-            throw new TrinoException(HUDI_CANNOT_OPEN_SPLIT, e);
-        }
+            if (fileStatus.isDirectory()) {
```

Comment:
Use `isFile()` since HDFS supports symlinks -- so `isDirectory()` is not the inverse of `isFile()`

Reply:
I didnt modify this, but is copied as-is from the previous code. Did you mean we need to now add `isFile` to HudiFileStatus?  I wonder if the error below must be updated instead